ts/pages/governance/modal_vote.tsx

```diff
@@ -1,238 +1,192 @@
-                            text={errorMessage}
-                            onClose={this._onCloseError.bind(this)}
-                        />
-                    </StyledDialogContent>
-                </DialogOverlay>
-            </>
-        );
-    }
-    public _renderFormContent(): React.ReactNode {
-        switch (this.state.isWalletConnected) {
-            case true:
-                return this._renderVoteFormContent();
-            case false:
-            default:
-                return this._renderConnectWalletFormContent();
+        if (!account.address || isFetchingVotingPowerData) {
+            return;
         }
-    }
-    private _shareViaTwitterAsync(): void {
-        const { zeipId } = this.props;
-        const tweetText = encodeURIComponent(`I voted on ZEIP-${zeipId}! 🗳️#VoteWithZRX https://0x.org/vote`);
-        window.open(`https://twitter.com/intent/tweet?text=${tweetText}`, 'Share your vote', 'width=500,height=400');
-    }
-    private _renderConnectWalletFormContent(): React.ReactNode {
-        const { web3Wrapper } = this.state;
-        return (
-            <>
-                <ConnectForm
-                    web3Wrapper={web3Wrapper}
-                    onDismiss={this.props.onDismiss}
-                    onWalletConnected={this._onWalletConnected.bind(this)}
-                    onError={this._onError.bind(this)}
-                />
-            </>
-        );
-    }
-    private _renderVoteFormContent(): React.ReactNode {
-        const { currentBalance, selectedAddress, web3Wrapper, web3, providerEngine } = this.state;
+        setIsFetchingVotingPowerData(true);
+        fetchDelegatorData()
+            .then(() => setIsFetchingVotingPowerData(false))
+            .catch(err => {
+                setIsFetchingVotingPowerData(false);
+                logUtils.warn(err);
+                errorReporter.report(err);
+            });
+    }, [account.address, apiClient]);
+
+    const onToggleConnectWalletDialog = React.useCallback(
+        (open: boolean) => {
+            const dispatcher = new Dispatcher(dispatch);
+            dispatcher.updateIsConnectWalletDialogOpen(open);
+        },
+        [dispatch],
+    );
+
+    const [isSuccessful, setSuccess] = React.useState(false);
+    const onVoted = React.useCallback((voteInfo: VoteInfo) => {
+        setSuccess(true);
+        onVoteInfoReceived(voteInfo);
+    }, []);
+    const [errorMessage, setErrorMessage] = React.useState<string | undefined>(undefined);
+    const [isErrorModalOpen, setErrorModalOpen] = React.useState(false);
+    const onVoteError = React.useCallback((message: string) => {
+        setSuccess(false);
+        setErrorMessage(message);
+        setErrorModalOpen(true);
+    }, []);
+
+    const onModalDismiss = React.useCallback(() => {
+        setSuccess(false);
+        setErrorMessage(undefined);
+        setErrorModalOpen(false);
+        onDismiss();
+    }, []);
```

Comment:
I think these callbacks are missing some dependencies in the dependency array. It's strange that `yarn lint` didn't catch that.

Running `tslint --fix` on the file gives me this
```diff
diff --git a/ts/pages/governance/modal_vote.tsx b/ts/pages/governance/modal_vote.tsx
index bef2f2f..fe5ef46 100644
--- a/ts/pages/governance/modal_vote.tsx
+++ b/ts/pages/governance/modal_vote.tsx
@@ -85,7 +85,7 @@ export const ModalVote: React.FC<ModalVoteProps> = ({ zeipId, isOpen, onDismiss,
                 logUtils.warn(err);
                 errorReporter.report(err);
             });
-    }, [account.address, apiClient]);
+    }, [account.address, apiClient, isFetchingVotingPowerData, networkId, providerState.web3Wrapper]);
 
     const onToggleConnectWalletDialog = React.useCallback(
         (open: boolean) => {
@@ -96,10 +96,13 @@ export const ModalVote: React.FC<ModalVoteProps> = ({ zeipId, isOpen, onDismiss,
     );
 
     const [isSuccessful, setSuccess] = React.useState(false);
-    const onVoted = React.useCallback((voteInfo: VoteInfo) => {
-        setSuccess(true);
-        onVoteInfoReceived(voteInfo);
-    }, []);
+    const onVoted = React.useCallback(
+        (voteInfo: VoteInfo) => {
+            setSuccess(true);
+            onVoteInfoReceived(voteInfo);
+        },
+        [onVoteInfoReceived],
+    );
     const [errorMessage, setErrorMessage] = React.useState<string | undefined>(undefined);
     const [isErrorModalOpen, setErrorModalOpen] = React.useState(false);
     const onVoteError = React.useCallback((message: string) => {
@@ -113,7 +116,7 @@ export const ModalVote: React.FC<ModalVoteProps> = ({ zeipId, isOpen, onDismiss,
         setErrorMessage(undefined);
         setErrorModalOpen(false);
         onDismiss();
-    }, []);
+    }, [onDismiss]);
 
     const bigNumberFormat = {
         decimalSeparator: '.',
```

Reply:
weird, my `tslint` ignores it. Anything special needed?

Reply:
I think the file glob for the lint command may be incorrect. It only searches 1 level down in the filesystem for ts files.